### PR TITLE
numbat: 1.12.0 -> 1.13.0

### PR DESCRIPTION
**Pull request by xeals**
## Description of changes

https://github.com/sharkdp/numbat/releases/tag/v1.13.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by giomf** [comment]
Result of `nixpkgs-review pr 341128` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>numbat</li>
  </ul>
</details>

Looks good to me!

**Review by giomf — APPROVED** [review]

**Comment by donovanglover** [comment]
Tests fail on darwin

**Comment by Atemu** [comment]
@donovanglover does the result work in your manual testing? If so we could simply disable checks on darwin for now.

File: pkgs/by-name/nu/numbat/package.nix [hunk 1]
```diff
@@ -5,20 +5,28 @@
...
   };
 
-  cargoHash = "sha256-t6vxJ0UIQJILCGv4PO5V4/QF5de/wtMQDkb8gPtE70E=";
+  cargoHash = "sha256-rK9RPd/hww2F87l/dd14pB4izE58NuqaewYaqMimV1M=";
+
+  patches = [
+    # Evaluating the currencies module (caused by referencing any non-euro Money
+    # unit) requires network access. See sharkdp/numbat#560.
+    # Remove this when updating to 1.14.0.
+    ./test_xkcd_2812.patch
```
Comment (fabianhjr) [inline pkgs/by-name/nu/numbat/package.nix:28]: Wouldn't the upstream patch be preferable?

https://github.com/sharkdp/numbat/commit/272c2455136b6e8debc3003b22db18fba6442822.patch

Comment (giomf) [reply pkgs/by-name/nu/numbat/package.nix:28]: Good catch! Yes, I would definitely suggest taking the patch from upstream.

Comment (xeals) [reply pkgs/by-name/nu/numbat/package.nix:28]: I did request the change but didn't think about using it ahead of time. I'll fix that up.

Comment (fabianhjr) [reply pkgs/by-name/nu/numbat/package.nix:28]: https://github.com/sharkdp/numbat/commit/4756a1989ecdab35fd05ca18c721ed15d8cde2b1

Sorry pasted the wrong commit, this one is the one that got merged upstream

**Comment by Atemu** [comment]
What are we going to do about Darwin?

I think if we don't get feedback from a Darwin user on whether the application works fine with `doCheck = false` in the next few days, we'll have to mark it as broken.

**Review by fabianhjr — APPROVED** [review]

**Comment by fabianhjr** [comment]
Approve is regarding patching/in general for linux platforms, will wait on input from darwin contributors/maintainers

**Comment by Atemu** [comment]
@xeals could you set numbat to be broken on darwin? Then we can merge.

**Review by Atemu — APPROVED** [review]

**Review by giomf — APPROVED** [review]

**Comment by Atemu** [comment]
Result of `nixpkgs-review pr 341128` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>numbat</li>
  </ul>
</details>

**Pull request merged**